### ISSUE TITLE
When running the Usage Example, I get "RuntimeError: First call try_lock()"

### ISSUE THREAD

**Issue by tinue** [adafruit/Adafruit_CircuitPython_BitbangIO#9]
On a Raspberry Pi Zero W Rev 1.1, running Raspbian Buster with kernel Linux 4.19.97+, I get the error "RuntimeError: First call try_lock()" when I run the "Usage Example" of the readme file. This is with Python 3.7.3.

While I can do this as suggested, I am wondering if there is an error somewhere. Raspbian was freshly installed and no other libraries are present, so the pins that the sample uses should not be in use. I also tried with the pins that I really need `bitbangio.SPI(clock=board.D23, MOSI=board.D24)` but I get the same result.

**Comment by tinue** [comment]
To confirm: When I do lock it works. I guess either the example should be adapted, or the try_lock should be done inside of the spi.write() method. Please let me know if you prefer a pull request.

**Comment by makermelissa** [comment]
Yes, please submit a PR to update the example. Thanks.

**Comment by tinue** [comment]
Hmm, I would actually prefer to update the library. Reason is that the hardware SPI through Adafruit_Blinka does **not** require locking on spi.write(). If I want to use the same code for hardware and bitbang SPI it's getting a bit confusing.
Is there a particular reason for the lock?

**Comment by makermelissa** [comment]
It was based on the CircuitPython bitbangio module, which checks for locks on reading and writing so that only one chip uses it at a time, but it's probably not strictly necessary if using the chip select.

**Comment by ladyada** [comment]
@makermelissa its a good idea to keep the lock because of folks using threaded python code :)

**Comment by tannewt** [comment]
Also note that we don't recommend using the SPI bus directly. It's much better to use the BusDevice library with the bus to do the locking because it uses context managers to lock and unlock.

https://github.com/adafruit/Adafruit_CircuitPython_BusDevice

**Comment by tinue** [comment]
Thank you! This was going to be my next step because I want to use chip select as well. In my first experiments I ignored this.

Edit: Looks like I'll continue to use SPI directly. The BusDevice is several orders of magnitude slower compared to using SPI directly. I need the roughly 6-8MHz that SPI offers.

Edit2: My mistake. I did set the baud rate on SPI before creating the BusDevice. If I do it correctly in the constructor of BusDevice then there is no speed difference.

Edit3: Wrong again... This call is much slower compared to using SPI directly:
```
with self.spibus as bus_device:
            bus_device.write(self.leds)
```
The write operation needs some orders of magnitude more time compared to SPI.write().
